core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/KeyStoreRegistryPersistenceManager.java

```diff
@@ -0,0 +1,373 @@
+
+        try {
+            // Create the public key resource.
+            Resource pubKeyResource = registry.newResource();
+            pubKeyResource.setContent(publicCert.getEncoded());
+            pubKeyResource.addProperty(PROP_TENANT_PUB_KEY_FILE_NAME_APPENDER, generatePublicCertId());
+            registry.put(RegistryResources.SecurityManagement.TENANT_PUBKEY_RESOURCE, pubKeyResource);
+
+            // Associate the public key with the keystore.
+            registry.addAssociation(RegistryResources.SecurityManagement.KEY_STORES + "/" + keyStoreName,
+                    RegistryResources.SecurityManagement.TENANT_PUBKEY_RESOURCE, ASSOCIATION_TENANT_KS_PUB_KEY);
+        } catch (RegistryException | CertificateEncodingException e) {
+            throw new SecurityException("Error when writing the keystore public cert to registry for keystore: " +
+                    keyStoreName, e);
+        }
+    }
+
+    /**
+     * Generates an ID for the public certificate, which is used as a file name suffix for the certificate.
+     * e.g. If keystore name is 'example-com.jks', public cert name will be 'example-com-343743.cert'.
+     *
+     * @return generated id to be used as a file name appender.
+     */
+    private String generatePublicCertId() {
+
+        String uuid = UUIDGenerator.getUUID();
+        return uuid.substring(uuid.length() - 6, uuid.length() - 1);
+    }
+
+    /**
+     * Delete the key store from the registry.
+     *
+     * @param keyStoreName Name of the key store.
+     * @throws SecurityException If an error occurs while deleting the key store.
+     */
+    public void deleteKeyStore(String keyStoreName) throws SecurityException {
+
+        String path = RegistryResources.SecurityManagement.KEY_STORES + REGISTRY_PATH_SEPARATOR + keyStoreName;
+        try {
+            org.wso2.carbon.registry.api.Association[] assocs = registry.getAllAssociations(path);
+            if (assocs.length > 0) {
+                throw new SecurityException("Key store : " + keyStoreName + " is already in use and can't be deleted");
+            }
+            registry.delete(path);
+        } catch (org.wso2.carbon.registry.api.RegistryException e) {
+            throw new SecurityException("Error deleting key store : " + keyStoreName, e);
+        }
+    }
+
+    /**
+     * Update the key store in the registry.
+     *
+     * @param keyStoreName Key store name.
+     * @param keyStore     Updated key store.
+     */
+    public void updateKeyStore(String keyStoreName, KeyStore keyStore) {
+
+        String path = RegistryResources.SecurityManagement.KEY_STORES + REGISTRY_PATH_SEPARATOR + keyStoreName;
+        CryptoUtil cryptoUtil = CryptoUtil.getDefaultCryptoUtil();
+        try (ByteArrayOutputStream outputStream = new ByteArrayOutputStream()) {
+            Resource resource = registry.get(path);
+            String encryptedPassword = resource
```

Comment:
Use `keyStoreRegistryPersistenceManager.getEncryptedKeyStorePassword` method

Reply:
Fixed in the latest changes

Reply:
Fixed in the latest changes